modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItInternalTableTest.java

```diff
@@ -384,78 +359,190 @@ public void testRoScanAllExplicitPopulatingData() throws InterruptedException {
+
+        InternalTable internalTable = ((TableImpl) table).internalTable();
+        var schemaDescriptor = ((TableImpl) table).schemaView().schema();
+
+        List<BinaryRow> res = internalTable.deleteAllExact(rowsToLookup, null).get();
+
+        var resIter = res.iterator();
+
+        for (BinaryRowEx key : rowsToLookup) {
+            int i = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, key)).<Long>value("key").intValue();
+
+            if (i % 2 == 1) {
+                var rowTuple = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, resIter.next()));
+
+                assertEquals(i % 100L, rowTuple.<Long>value("key"));
+                assertEquals(i, rowTuple.<Integer>value("valInt"));
+                assertEquals("some string row" + i, rowTuple.<Integer>value("valStr"));
             }
+        }
+    }
+
+    @Test
+    public void insertAllOrderTest() throws Exception {
+        var rowsToLookup = populateEvenKeysAndPrepareEntriesToLookup(false);
+
+        InternalTable internalTable = ((TableImpl) table).internalTable();
+        var schemaDescriptor = ((TableImpl) table).schemaView().schema();
+
+        List<BinaryRow> res = internalTable.insertAll(rowsToLookup, null).get();
+
+        var resIter = res.iterator();
 
-            @Override
-            public void onError(Throwable throwable) {
-                fail("onError call is not expected.");
+        for (BinaryRowEx key : rowsToLookup) {
+            int i = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, key)).<Long>value("key").intValue();
+
+            if (i % 2 == 0) {
+                var rowTuple = TableRow.tuple(Row.wrapBinaryRow(schemaDescriptor, resIter.next()));
+
+                assertEquals(i % 100L, rowTuple.<Long>value("key"));
+                assertEquals(i, rowTuple.<Integer>value("valInt"));
+                assertEquals("some string row" + i, rowTuple.<Integer>value("valStr"));
             }
+        }
+    }
+
+    private ArrayList<BinaryRowEx> populateEvenKeysAndPrepareEntriesToLookup(boolean keyOnly) {
+        KeyValueView<Tuple, Tuple> keyValueView = table.keyValueView();
+
+        var keyRows = new ArrayList<BinaryRowEx>();
 
-            @Override
-            public void onComplete() {
-                subscriberAllDataAwaitLatch2.countDown();
+        for (int i = 0; i < 15; i++) {
+            keyRows.add(keyOnly ? createKeyRow(i) : createKeyValueRow(i, i, "some string row" + i));
+
+            if (i % 2 == 0) {
+                putValue(keyValueView, i);
             }
-        });
+        }
+
+        Collections.shuffle(keyRows);
 
-        subscriberAllDataAwaitLatch2.await();
+        return keyRows;
+    }
+
+    private void roScanAll(boolean implicit) throws InterruptedException {
+        IgniteImpl node = node();
+
+        KeyValueView<Tuple, Tuple> keyValueView = table.keyValueView();
+
+        var retrievedItems = scanAllPartitions(node);
```

Comment:
It is incorrect to use `var` because the type is not visible from the line of code.